llama_index/node_parser/interface.py

```diff
@@ -60,18 +60,30 @@ def get_nodes_from_documents(
         ) as event:
             nodes = self._parse_nodes(documents, show_progress=show_progress, **kwargs)
 
-            if self.include_metadata:
-                for node in nodes:
-                    if (
-                        node.ref_doc_id is not None
-                        and node.ref_doc_id in doc_id_to_document
-                    ):
+            for i, node in enumerate(nodes):
+                if (
+                    node.ref_doc_id is not None
+                    and node.ref_doc_id in doc_id_to_document
+                ):
+                    ref_doc = doc_id_to_document[node.ref_doc_id]
+                    start_char_idx = ref_doc.text.find(
+                        node.get_content(metadata_mode=MetadataMode.NONE)
+                    )
+
+                    # update start/end char idx
+                    if start_char_idx >= 0:
+                        node.start_char_idx = start_char_idx
```

Comment:
if we don't update start and end idx, what will happen

Reply:
it will stay none (which was the previous behaviour, it was always none)